skbeam/core/correlation.py

```diff
@@ -769,3 +769,70 @@ def _validate_and_transform_inputs(num_bufs, num_levels, labels):
 
     return (label_array, pixel_list, num_rois, num_pixels,
             lag_steps, buf, img_per_level, track_level, cur)
+
+
+
+def get_four_time_from_two_time(  g12,g2=None, rois=None  ):
+    ''' 
+    Dec 16, 2015, Y.G.@CHX
+    Get four-time correlation function from two correlation function
+    namely, calculate the deviation of each diag line of g12 to get four-time correlation fucntion
+    TOBEDONE: deal with bad frames
+    
+    Parameters:
+        g12: a 3-D array, two correlation function, shape as ( imgs_length, imgs_length, q)  
+    
+    Options:
+        g2: if not None, a 2-D array, shape as ( imgs_length,  q), or (tau, q)
+            one-time correlation fucntion, for normalization of the four-time
+        rois: if not None, a list, [x-slice-start, x-slice-end, y-slice-start, y-slice-end]
+   
+    Return:
+        g4f12: a 2-D array, shape as ( imgs_length,  q), 
+                   a four-time correlation function  
+     
+    One example:        
+        s1,s2 = 0,2000
+        g4 = get_four_time_from_two_time( g12bm, g2b, roi=[s1,s2,s1,s2] )
+         
+    '''      
+    
+    
+    m,n,noqs = g12.shape
+    g4f12 = []       
+    for q in  range(noqs):   
```

Comment:
If you make the above suggested change to the inputh shape, this can be re-writen as `for y in g12:`